packages/editor/src/components/commands/index.js

```diff
@@ -338,6 +354,32 @@ const getEditedEntityContextualCommands = () =>
 				},
 			} );
 		}
+		if ( isViewable ) {
+			if ( postType === 'post' || postType === 'page' ) {
```

Comment:
I appreciate your work on this @benazeer-ben Thank you!

Sorry, this wasn't known to me before, but I think a command already exists to **preview** in the editor:

https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/commands/index.js#L275

Here they are together:

<img width="449" alt="Screenshot 2024-12-19 at 11 37 12 am" src="https://github.com/user-attachments/assets/b9e6c30d-5e3e-4a49-ae93-08506cd4609f" />

It's been there for some time. 

I think this PR's command adds a little sugar by checking post status and having a "view" link, but the existing command will work for all posts, plus it also saves before previewing using `__unstableSaveForPreview`.

So, with this knowledge, maybe we need to combine. One option would be to rather use the existing command, and add a generic `View in a new tab` if the post is published (so don't gate for pages or posts). I'm not sure.

cc @jameskoster @youknowriad (who added the existing command)

Reply:
@ramonjd 

This PR adds some useful refinement by incorporating checks for the post status and providing a "View" link specifically for published posts, which could serve as a valuable enhancement.

With this in mind, rather than combining the commands, would it make sense to focus solely on the 'View' functionality in this PR and omit the preview option? This could simplify the implementation while keeping the focus clear.

I’d love to hear thoughts from other reviewers on this approach as well.

Thanks again for highlighting this!

Reply:
> With this in mind, rather than combining the commands, would it make sense to focus solely on the 'View' functionality in this PR and omit the preview option? 

I think that's a good compromise for the first iteration. 

> 'Preview' could be about checking a draft, or unsaved changes to a published record. 'View' could be about viewing a published record ignoring any unsaved changes.

Does that imply that we'd gate the [existing preview command](https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/commands/index.js#L275) with a `getEditedPostAttribute( 'status' )` check (not published), similar to what @benazeer-ben has done in this PR?

Then the changes in PR could be reduced in scope to editing a published record [as mentioned](https://github.com/WordPress/gutenberg/pull/66720#discussion_r1891192101).